tests/testthat/ressources/with-abort-error.Rmd

```diff
@@ -0,0 +1,15 @@
+---
+title: document with error
+---
+
+```{r}
+rlang::global_entrace()
+options(rlang_backtrace_on_error_report = "full")
```

Comment:
Do we need these lines?

Reply:
I think we do need `global_entrace()` until I fix it in **knitr**. **evaluate** won't entrace error by itself and my understanding is that the way **knitr** does it is currently not compatible with new restart logic of evaluate. I'll check it though, but I have a fix in for **knitr** already. 

`options(rlang_backtrace_on_error_report = "full")` is not needed though. It is only useful when error is not thrown and kept in cell output (e.g `error = TRUE`), and we don't need to test this IMO.

Reply:
Oh I did not read well... THis is in abort version ! We don't need it indeed.

Reply:
Maybe we also need a stop version with out it, just to show the difference?

Reply:
yes, that was a good think to check that. 

I think this PR now also solve the problem with **knitr** I mentioned above. 

`rlang::global_entrace()` is not required anymore as **knitr** will do `rlang::entrace()`. We added that in https://github.com/yihui/knitr/pull/2218 some time ago. 

This is now enough to have entraced error when **rlang** is installed 
````markdown
---
title: document with error
---

```{r}
f <- function() g()
g <- function() h()
h <- function() stop("!")
f()
```
````

`rlang::global_entrace()` is needed when `error = TRUE` so that the error are capturing for cell output is entraced. 
````markdown
---
title: document with error
---

```{r}
rlang::global_entrace()
options(rlang_backtrace_on_error_report = "full")
```

```{r, error = TRUE}
f <- function() g()
g <- function() h()
h <- function() stop("!")
f()
```
````

If we want this to also be automatically done by **knitr**, this is where I need to fix it. 

I'll see how to adapt the test to get all the case. But maybe part of it is to test at **knitr** level, and for **evaluate** to only test its own logic (without **rmarkdown** in the test)